my_2d_robot/src/joint_states_subscriber.cpp

```diff
@@ -1,142 +1,9 @@
+#include "joint_states_subscriber.h"
```

Comment:
I think I understand. 
So I need to:
- move the function definitions to the joint_state_subscriber.cpp file and 
- create a main.cpp for this node where all the function calls are done? 

However, since the callback function calls the functions to calculate the end effector position and to save the end effector, then the only function call in the main.cpp file will be the one for the constructor, similar to the current joint_states_subscriber.cpp?

Reply:
Precisely!